### PR TITLE
(PDB-1099) Rename catalog :name key to :certname

### PR DESCRIPTION
**Pull request by mullr**
For consistency with other endpoints. This affects responses from the
/v4/catalogs endpoint, queries to it, and the "replace catalog" command.

### CODE REVIEW

**Comment by mullr** [comment]
@pljenkinsro okay to test

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

File: documentation/api/commands.markdown [hunk 1]
```diff
@@ -69,27 +69,30 @@ processed.
...
 payload of this command.
 
 ### "deactivate node", version 2
 
-The payload is expected to be the name of a node, as a raw JSON string, which will be deactivated
-effective as of the time the command is *processed*. Serialization of the payload is no
-longer required.
+The payload is expected to be the certname of a node, as a raw JSON
+string, which will be deactivated effective as of the time the command
+is *processed*. Serialization of the payload is no longer required.
```
Comment (wkalt) [inline documentation/api/commands.markdown]: not your change, but I think we should drop the last sentence here. We've dropped references to previous versions in other spots so "no longer" just invites confusion.

File: src/puppetlabs/puppetdb/query/catalogs.clj [hunk 2]
```diff
@@ -91,9 +91,11 @@
                    (filter #(not (nil? (:source_type %))))
                    (reduce collapse-edges #{})
                    (into []))]
-    (assoc (select-keys first-row [:name :version :environment :hash
+    (assoc (select-keys first-row [:version :environment :hash
                                    :transaction_uuid :producer_timestamp])
-      :edges edges :resources resources)))
+           :certname (:certname first-row)
```
Comment (wkalt) [inline src/puppetlabs/puppetdb/query/catalogs.clj]: we can put the :certname key in the select-keys call

**Comment by wkalt** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by wkalt** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
﻿

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
^^ transient errors, running again. 

@pljenkinsro retest this please

File: src/puppetlabs/puppetdb/scf/storage.clj [hunk 3]
```diff
@@ -701,7 +701,7 @@
          hash           (time! (:catalog-hash metrics)
                                (shash/catalog-similarity-hash catalog))
          {id :id
-          stored-hash :hash} (catalog-metadata name)]
+          stored-hash :hash} (catalog-metadata certname)] ;; TODO: shouldn't this be done with a join to certnames?
```
Comment (senior) [inline src/puppetlabs/puppetdb/scf/storage.clj]: Can you explain more here? Is this something we need to change?

**Comment by senior** [comment]
+1 from me, just a question about the TODO comment that was added

**Comment by mullr** [comment]
@senior That was just detritus from my first pass through the code, probably based on a misunderstanding of what was going on... because it makes no sense. Removed it.

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by senior** [comment]
Sounds good, will merge this once master is clear

**Comment by mullr** [comment]
@pljenkinsro retest this please

**Comment by wkalt** [comment]
@pljenkinsro retest this please

**Comment by mullr** [comment]
:dancer:

**Pull request merged**